static/js/components/CourseCarousel.js

```diff
@@ -16,50 +17,66 @@ type Props = {|
   setShowResourceDrawer: Function
 |}
 
-const prevButton = ({ previousSlide, currentSlide }) =>
-  currentSlide === 0 ? null : (
-    <div onClick={previousSlide} className="carousel-control prev">
-      <i className="material-icons">chevron_left</i>
-    </div>
-  )
+export default function CourseCarousel(props: Props) {
+  const { title, courses, setShowResourceDrawer } = props
 
-const nextButton = props => {
-  const { slideCount, nextSlide, currentSlide, slidesToShow } = props
-  const lastSlideIndex = slideCount - 1
+  const [index, setIndex] = useState(0)
+  const canPageUp = index + CAROUSEL_PAGE_SIZE < courses.length
+  const canPageDown = index !== 0
 
-  const disabled = currentSlide + slidesToShow > lastSlideIndex
-  return disabled ? null : (
-    <div onClick={nextSlide} className="carousel-control next">
-      <i className="material-icons">chevron_right</i>
+  return (
+    <div className="course-carousel">
+      <div className="title-and-controls">
+        <div className="title">{title}</div>
+        <div className="controls">
+          <button
+            className={`dark-outlined compact extra-compact prev ${
+              canPageDown ? "" : "disabled"
+            }`}
+            disabled={!canPageDown}
+            onClick={
+              canPageDown
+                ? () => setIndex(R.max(index - CAROUSEL_PAGE_SIZE, 0))
+                : null
+            }
+          >
+            <i className="material-icons">arrow_back</i>
+            Previous
+          </button>
+          <button
+            className={`dark-outlined compact extra-compact next ${
+              canPageUp ? "" : "disabled"
+            }`}
+            onClick={
+              canPageUp ? () => setIndex(index + CAROUSEL_PAGE_SIZE) : null
+            }
+            disabled={!canPageUp}
+          >
+            Next
+            <i className="material-icons">arrow_forward</i>
+          </button>
+        </div>
+      </div>
+      <Carousel
+        slideIndex={index}
+        slidesToShow={CAROUSEL_PAGE_SIZE}
+        slidesToScroll="auto"
+        withoutControls={true}
+        afterSlide={slideIndex => setIndex(slideIndex)}
```

Comment:
Is setting `afterSlide` necessary? The carousel seemed to behave the same with or without this line in some brief testing.

Reply:
the `afterSlide` is what keeps the `slideIndex` in sync with the position of the carousel. The carousel has it's own internal state which manages which page it's showing - here basically I'm replicating that state externally (the `index` variable) so that I can control that state with the buttons. Then I pass that state down into the carousel (via the `slideIndex` prop) to control the carousel and sync and carousel-side changes to the position back to the `index` via this `afterSlide` callback.

basically, the case where it can get out of sync is if the user slides the carousel. if we're not syncing that change in the index back (via `afterSlide`) and the user then goes to click the button, we'll change teh `index` based on an incorrect current value for the index (b/c it hasn't been updated to reflect the changed position of the carousel) and we'll get some weird behavior.

does that make sense? I can show you how it can lead to weird behavior if that helps too

Reply:
sounds good